File: backend/src/tests/secret.rs
use crate::secret::{Secret, SecretDownloadPolicy, SecretTTL};
use crate::tests::get_random_string;

pub fn get_sample_secret() -> Secret {
    Secret {
        id: get_random_string(),
        payload: get_random_string(),
        ttl: SecretTTL::OneHour,
        download_policy: SecretDownloadPolicy::Unlimited
    }
}

File: frontend/src/lib/secret.ts
export class Secret {
	id: string = '';
	payload: string = '';
	contentType: SecretContentType = SecretContentType.Text;
	ttl: SecretTTL = SecretTTL.OneHour;
	downloadPolicy: SecretDownloadPolicy = SecretDownloadPolicy.OneTime;
	metadata?: FileMetadata;
}

export enum SecretContentType {
	Text = 'Text',
	File = 'File'
}

export enum SecretTTL {
	OneHour = 'OneHour',
	TwoHours = 'TwoHours',
	OneDay = 'OneDay',
	OneWeek = 'OneWeek'
}

export enum SecretDownloadPolicy {
	OneTime = 'OneTime',
	Unlimited = 'Unlimited'
}

export class FileMetadata {
	name: string = '';
	type: string = '';
	size: number = 0;
}
